### PR TITLE
Fix `line_plot()` to work with `nan` in meta columns

### PR DESCRIPTION
**Pull request by danielhuppmann**
# Please confirm that this PR has done the following:

- [x] Tests Added (extended existing `line_plot()` test with metadata, first failed, then passed)
- [x] Documentation Added
- [x] Description in RELEASE_NOTES.md Added

# Description of PR

The function `line_plot()` drops all rows where any value is `nan` in the data or meta tables, even if the columns in questions aren't used. This PR adds a function `join_meta()` where only columns (in particular those relevant to the plotting library) are joined to the dataframe which is passed to the plotting functions.

### CODE REVIEW

**Comment by gidden** [comment]
Nice! I'm a bit curious how you see the use of this `join_meta()` function. 

Is it something that "normal/every day" users will want with some regularity? How should it be used vis-a-vis `as_pandas()`? Should it replace it? Should it be used with the `with_metadata` option? 

I think now that the package is getting more mature (and bigger), we should try to avoid code bloat wherever possible and reuse as much as possible. Thus my intuition says that `as_pandas` should probably use this functionality somehow.

**Comment by danielhuppmann** [comment]
As discussed offline with @gidden, I added the feature for column discovery to the `as_pandas()` function, so if `with_metadata` is a dictionary, it will only merge meaningful columns, keeping the existing API unchanged.

Also, if `with_metadata=False`, the function now returns a copy of the `data` rather than the data frame itself.

**Comment by gidden** [comment]
looks like one of the win ci builds failed on an http request, rebuilding

File: pyam/core.py [hunk 1]
```diff
@@ -1079,7 +1068,7 @@ def line_plot(self, x='year', y='value', **kwargs):
 
         see pyam.plotting.line_plot() for all available options
         """
-        df = self.join_meta(x=x, y=y, **kwargs)
+        df = self.as_pandas(with_metadata=kwargs)
```
Comment (gidden) [inline pyam/core.py]: do we need to include `x` and `y` here? or no?

Comment (danielhuppmann) [reply pyam/core.py]: I don't think we do - if `x` or `y` are variables, they will be included in the returned dataframe anyway. And I don't think we can do `x` or `y` as metadata columns currently?

**Comment by gidden** [comment]
thanks @danielhuppmann !

**Pull request merged**